index.html

```diff
@@ -533,6 +516,29 @@ <h2>
           going to change at all.
         </p>
       </section>
+      <section>
+        <h2>
+          Common Safety Checks
+        </h2>
+        <p>
+          The <dfn>common safety checks</dfn> for a {{Document}}
+          |document:Document| are the following steps:
+        </p>
```

Comment:
You should make these throw an exception. Web IDL will do the conversion from a thrown exception to a rejected promise. Returning an exception is very weird.

Alternatively you could return a(n) (string) enum and branch on that.

Reply:
Ok, yeah, good point... will make them throw directly.

Reply:
Ok, sooooo... these all throw now. And yes, the IDL layer would automatically wrap the exceptions. 

However, just from the sake of clarity and consistency, I'd still like to use "Return a promise rejected with X" for `.lock()`. I've had instances in the past of reviewers getting confused about methods throwing even though they always return a promise. 

Similarly, for .unlock(), just for clarity, I have it re-throw and abort the steps.

Reply:
I guess that's okay, but I'd rather we explain the mechanism in a note so implementations with correct binding layers don't do this redundantly as well.

Reply:
Yeah, it would be way cleaner. I'd also rather just be writing: 

* If X throw {{WhateverError}}.
* If Y throw {{OtherError}}.

And so on... It feels like WebIDL's [=exception/throw=] should include such a note. Currently, it's not super clear what should happen and every example in WebIDL is using [=return a promise rejected with=]. 

I'll raise an issue on WebIDL.

Reply:
Anyone interested https://github.com/whatwg/webidl/issues/1301